src/markdown.c

```diff
@@ -1744,7 +1742,7 @@ parse_listitem(struct buf *ob, struct sd_markdown *rndr, uint8_t *data, size_t s
 			if (in_empty)
 				has_inside_empty = 1;
 
-			if (pre == orgpre) /* the following item must have */
+			if (i <= orgpre) /* the following item must have */
```

Comment:
It would be parsed as:

```
+ root (type 0, data 0) - `` :0:26
  + list item (type 6, data 0) - `` :0:5
    + paragraph (type 7, data 0) - `a
` :3:2
  + list item (type 6, data 0) - `` :5:21
    + paragraph (type 7, data 0) - `b
` :7:2
    + list item (type 6, data 0) - `` :13:5
      + paragraph (type 7, data 0) - `c
` :16:2
    + list item (type 6, data 0) - `` :22:4
      + paragraph (type 7, data 0) - `d
` :24:2
```

Which is correct.